runtime/bcutil/cfreader.c

```diff
@@ -2047,15 +2047,50 @@ checkAttributes(J9CfrClassFile* classfile, J9CfrAttribute** attributes, U_32 att
 				}
 				bootstrapMethodAttributeRead = TRUE;
 				for (j = 0; j < bootstrapMethods->numberOfBootstrapMethods; j++) {
-					value = bootstrapMethods->bootstrapMethods[j].bootstrapMethodIndex;
-					if((0 == value) || (value > cpCount)) {
+					U_16 numberOfBootstrapArguments = 0;
+					J9CfrBootstrapMethod *bsm = &bootstrapMethods->bootstrapMethods[j];
+					value = bsm->bootstrapMethodIndex;
+					if ((0 == value) || (value > cpCount)) {
 						errorCode = J9NLS_CFR_ERR_BAD_INDEX__ID;
 						goto _errorFound;
 					}
-					if(cpBase[value].tag != CFR_CONSTANT_MethodHandle) {
+					if (cpBase[value].tag != CFR_CONSTANT_MethodHandle) {
 						errorCode = J9NLS_CFR_ERR_BOOTSTRAP_METHODHANDLE__ID;
 						goto _errorFound;
 					}
+
+					numberOfBootstrapArguments = bsm->numberOfBootstrapArguments;
+					for (k = 0; k < numberOfBootstrapArguments; k++) {
+						U_8 cpValueTag = 0;
+						value = bsm->bootstrapArguments[k];
+
+						if ((0 == value) || (value > cpCount)) {
+							errorCode = J9NLS_CFR_ERR_BAD_INDEX__ID;
+							goto _errorFound;
+						}
+						/* Validate the constant_pool indexes stored in the bootstrap_arguments array.
+						 * Note: The constant_pool entry at that index must be a CONSTANT_String_info,
+						 * CONSTANT_Class_info, CONSTANT_Integer_info, CONSTANT_Long_info, CONSTANT_Float_info,
+						 * CONSTANT_Double_info, CONSTANT_MethodHandle_info, or CONSTANT_MethodType_info structure.
+						 */
+						cpValueTag = cpBase[value].tag;
+						switch(cpValueTag) {
+						case CFR_CONSTANT_String:
+						case CFR_CONSTANT_Class:
+						case CFR_CONSTANT_Integer:
+						case CFR_CONSTANT_Long:
+						case CFR_CONSTANT_Float:
+						case CFR_CONSTANT_Double:
+						case CFR_CONSTANT_MethodHandle:
+						case CFR_CONSTANT_MethodType:
+						case CFR_CONSTANT_Dynamic:
```

Comment:
This assumes that Constant_Dynamics will be rejected in other places prior to JDK11?

Reply:
Technically yes as such check is already done in readPool() before the check of BSM.
```
readPool(J9CfrClassFile* classfile, U_8* data, U_8* dataEnd, U_8* segment, U_8* segmentEnd, U_8** pIndex, U_8** pFreePointer)
{
...
		case CFR_CONSTANT_Dynamic:
			if (classfile->majorVersion < 55) {
				errorCode = J9NLS_CFR_ERR_CP_ENTRY_INVALID_BEFORE_V55__ID;
				offset = (U_32) (index - data - 1);
				goto _errorFound;
			}
```
So there is no need to duplicate the same check here.